.azure/pr.yml

```diff
@@ -0,0 +1,52 @@
+pr:
+  autoCancel: true
+  branches:
+    include: ["*"]
+
+stages:
+  # The preliminary stage should be small in both total runtime (including
+  # provisioning) and resources required.  About half of PR commits result in a
+  # CI failure, and over 90% of these are in linting, documention or a test
+  # failure that would affect _any_ OS or Python version.  The goal in the first
+  # stage is to catch the vast majority of failures with minimal cost.
+  - stage: "Lint_Docs_Prelim_Tests"
+    jobs:
+      - template: "templates/lint-linux.yml"
+        parameters:
+          pythonVersion: "3.7"
+
+      - template: "templates/docs-linux.yml"
+        parameters:
+          pythonVersion: "3.7"
+
+      - template: "templates/test-linux.yml"
+        parameters:
+          pythonVersions: ["3.7"]
+          testQPY: false
+          testImages: false
+
+  # The rest of the PR pipeline is to test the oldest and newest supported
+  # versions of Python, along with the integration tests (via the tutorials).
+  # It's very rare for a failure to be specific to an intermediate version of
+  # Python, so we just catch those in the cron-job pipeline to reduce the amount
+  # of resources used.
+  - stage: "Tutorials_and_Tests"
+    dependsOn: "Lint_Docs_Prelim_Tests"
+    jobs:
+      - template: "templates/tutorials-linux.yml"
+        parameters:
+          pythonVersion: "3.8"
```

Comment:
Is there a specific reason for 3.8?

Reply:
Only continuity, nothing else - this is what it we currently use.